vpr/src/route/route_timing.cpp

```diff
@@ -1669,6 +1675,107 @@ static void print_route_status(int itry, double elapsed_sec, float pres_fac, int
     fflush(stdout);
 }
 
+static void print_overused_nodes_status(const t_router_opts& router_opts, const OveruseInfo& overuse_info) {
+    //Display upper limit
+    if (int(overuse_info.overused_nodes) > router_opts.max_reported_overused_rr_nodes) {
+        return;
+    }
+
+    //Print overuse info header
+    print_overused_nodes_header();
+
+    //Print overuse info body
+    const auto& device_ctx = g_vpr_ctx.device();
+    const auto& route_ctx = g_vpr_ctx.routing();
+
+    int overuse_index = 0;
+    for (size_t inode = 0; inode < device_ctx.rr_nodes.size(); inode++) {
+        int overuse = route_ctx.rr_node_route_inf[inode].occ() - device_ctx.rr_nodes[inode].capacity();
+
+        if (overuse > 0) {
+            print_single_overused_node_status(overuse_index, inode);
+            overuse_index++;
+        }
+    }
+
+    VTR_LOG("Total number of overused nodes: %d\n", overuse_info.overused_nodes);
+    VTR_LOG("\n");
+}
+
+static void print_overused_nodes_header() {
+    VTR_LOG("\nRouting Failure Diagnostics: Printing Overused Nodes Information\n");
+    VTR_LOG("------ ------- ---------- --------- -------- ------------ ------- ------- ------- ------- ------- ------- ------------ -----------------\n");
+    VTR_LOG("   No.   Inode  Occupancy  Capacity  RR Node    Direction    Side     PTC    Xlow    Ylow   Xhigh   Yhigh   Resistance       Capacitance\n");
+    VTR_LOG("                                        type                          NUM                                                               \n");
+    VTR_LOG("------ ------- ---------- --------- -------- ------------ ------- ------- ------- ------- ------- ------- ------------ -----------------\n");
+}
+
+static void print_single_overused_node_status(int overuse_index, int inode) {
```

Comment:
I would suggest using the strong id of rr_node `RRNodeId` because integer id may be misused easily. People may give an invalid integer which crashes VPR. This will be compatible with the refactoring effort on rr_graph